### ISSUE TITLE
Is this project affected by the new gtag.js implementation?

### ISSUE THREAD

**Issue by rfgamaral** [jehna/ga-lite#47]
Google recently started recommending one to track using gtag.js instead of analytics.js and I'm curious how this affects this project, if it affects it at all.

More information: https://developers.google.com/analytics/devguides/collection/gtagjs/migration

**Comment by jehna** [comment]
Yep, ga-lite imitates the older analytics.js. Perhaps in the future this project will transition towards using the new kind of API, but this is not planned yet.

Universal Analytics (gtag.js) offers a few upgrades to the previous Classic Analytics (analytics.js):
- Custom variables => custom dimensions and metrics
- Offline-enabled data sync
- Android and iOS libraries

**Comment by rfgamaral** [comment]
If you're not closing that label might be a bit misleading though.

Anyway, from your point of view, if one doesn't use those new features, is there no issue in using this lib of yours? All my analytics will still be tracked as if I was using the original version?

Still, is the resume installation still the recommend one? I'm sure I've seen a blog post from you with a different, shorter implementation. Not sure if it was an older or more recent implementation.

**Comment by frank6tg** [comment]
My experience is that it works perfectly (I've used it on about 5 sites). I
have had to re-validate ownership on Google Search Console (Webmaster
Tools) using another method such as an html file instead of using analytics.


Respectfully.

Frank Gomez
941-374-7711
Owner / Creative Director
Empowered Marketing, LLC dba Frank's Designs
Empowered Marketing™
<http://www.franksdesigns.com/>www.FranksDesigns.com
<http://www.franksdesigns.com/>
www.WPTechGuru.com

**Comment by jehna** [comment]
@rfgamaral 

> if one doesn't use those new features, is there no issue in using this lib of yours? All my analytics will still be tracked as if I was using the original version?

All analytics (other than #46) are tracked normally. I'm using this in my own projects and it works just fine. You might see *some* differences in the amount of absolute visitors/events when migrating, because some ad blockers do block analytics.js, but they don't necessarily block ga-lite. But Google Analytics numbers are not absolutely accurate anyways (I'm generally seeing 20-50% less page views at Google Analytics than in my server logs because of the popularity of ad blockers, regardless of if I'm using ga-lite or the official analytics).

> Still, is the resume installation still the recommend one? I'm sure I've seen a blog post from you with a different, shorter implementation. Not sure if it was an older or more recent implementation.

The one in README is the most up to date. The 1.x version had a shorter snippet (which is still available from npm/jsdelivr if you liked the old syntax) and it is not supported anymore in the 2.x version. The new 2.0 version introduced many new features, including event tracking, so you might want to use it if you're doing anything else than page view tracking with your Google Analytics.

**Comment by frank6tg** [comment]
I just noticed something - the note on migrating from v1 - it is fairly
important because it affects your bounce rate.


Respectfully.

Frank Gomez
941-374-7711
Owner / Creative Director
Empowered Marketing, LLC dba Frank's Designs
Empowered Marketing™
<http://www.franksdesigns.com/>www.FranksDesigns.com
<http://www.franksdesigns.com/>
www.WPTechGuru.com

**Comment by rfgamaral** [comment]
Well, to be honest, I decided to just conditionally load GA if not being crawled by Google PageSpeed Insights: https://github.com/rfgamaral/ricardoamaral.net/commit/a1613798f1c1434c571fc76ad554a3c324e471a9

**Comment by jamesjohnmcguire** [comment]
I've switched over to using the gtag format for one of my clients, as they are now running multiple tracking campaigns, so the final bits of JavaScript are something like:
	gtag('config', 'UA-XXXXXXXX-1');
	gtag('event', 'conversion',	{'send_to': 'YYYYYYY-YYYYYY});

It isn't clear to me, if ga-lite supports this or how to go about implementing it.

**Comment by jehna** [comment]
@jamesjohnmcguire Seems that you're using Google Ads on your site. This piece of code is responsible for sending the conversion information for Google Ads:

> `gtag('event', 'conversion', {'send_to': 'YYYYYYY-YYYYYY});`
> (See more: https://support.google.com/google-ads/answer/7548399?hl=en)

ga-lite unfortunately only supports Google Analytics; this project does not support any other Google Tag Manager related projects like Google Ads.
